wandb/sdk/launch/builder/kaniko.py

```diff
@@ -363,13 +384,21 @@ def _create_kaniko_job(
                 args=args,
                 volume_mounts=volume_mounts,
                 env=[env],
+                resources=client.V1ResourceRequirements(
+                    requests={"cpu": "1", "memory": "2Gi", "ephemeral-storage": "8Gi"},
+                    limits={"cpu": "2", "memory": "4Gi", "ephemeral-storage": "16Gi"},
+                ),
             )
         else:
             container = client.V1Container(
                 name="wandb-container-build",
                 image="gcr.io/kaniko-project/executor:v1.8.0",
                 args=args,
                 volume_mounts=volume_mounts,
+                resources=client.V1ResourceRequirements(
+                    requests={"cpu": "1", "memory": "2Gi", "ephemeral-storage": "8Gi"},
```

Comment:
I'm not sure that 2Gi is enough memory, have you tested this with environments that install a lot of packages?

Reply:
I guess I'm not sure what counts as a lot of packages? I tried with code that installs torch and a bunch of other ml libraries. I'm not even sure why I added a minimum cpu and memory ask, the reason I put this in was for the ephemeral storage request which was needed to keep builds from always failing on GKE autopilot. Ultimately builder requests and limits should be set in the agent config I think.

Reply:
I agree, rather than hard coding these we should have them be captured in the agent config and piped to the BuildConfig classes introduced by your new design...